From 56a1129102bfee95a1a8d04dc2374acf497784a2 Mon Sep 17 00:00:00 2001
From: aihuaxu <aihuaxu@apache.org>
Date: Thu, 19 Nov 2015 09:07:50 -0500
Subject: [PATCH 0396/1164] CDH-32512 HIVE-12418:
 HiveHBaseTableInputFormat.getRecordReader()
 causes Zookeeper connection leak. (Naveen Gangam
 via Aihua Xu)

Change-Id: I4a54e0b7cd95c2fd4760de3330e616c7ca35cecb
---
 .../hive/hbase/HiveHBaseTableInputFormat.java      |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/hbase-handler/src/java/org/apache/hadoop/hive/hbase/HiveHBaseTableInputFormat.java b/hbase-handler/src/java/org/apache/hadoop/hive/hbase/HiveHBaseTableInputFormat.java
index 5f4a1e4..a5bfb6d 100644
--- a/hbase-handler/src/java/org/apache/hadoop/hive/hbase/HiveHBaseTableInputFormat.java
+++ b/hbase-handler/src/java/org/apache/hadoop/hive/hbase/HiveHBaseTableInputFormat.java
@@ -116,6 +116,7 @@
       @Override
       public void close() throws IOException {
         recordReader.close();
+        closeTable();
       }
 
       @Override
@@ -511,4 +512,13 @@ static IndexPredicateAnalyzer newIndexPredicateAnalyzer(
       closeTable();
     }
   }
+
+  @Override
+  protected void finalize() throws Throwable {
+    try {
+      closeTable();
+    } finally {
+      super.finalize();
+    }
+  }
 }
-- 
1.7.9.5

